### DISCUSSION TITLE
Teleports

### DISCUSSION THREAD

**Discussion by mariomadproductions** [city41/smaghetti#79] [General]
I think I've worked out how the [teleports/warps in Castle Dash work](https://youtu.be/sJ386iNVTik?t=390). If you place an 01 00 sprite over a transport, the transport will act as an invisible teleport location (like in the NSMB games), which spans the height of the level. It does seem to work from the left (not just the right as used in Castle Dash), but I haven't thoroughly tested that.

**Comment by city41** [comment]
nice, that will be good to add to smaghetti. based on what you found doesn't sound too difficult :crossed_fingers:

**Comment by mariomadproductions** [comment]
Btw, is it possible to go back and forth between the user-friendly editor and the hex-based editor? (Or maybe could you add support for adding "undocumented" sprites using the user-friendly editor. I'm just thinking I might want to make a level using these teleports.

**Reply by city41** [reply]
currently not in a great way. you can create a level in smaghetti then load it in hex-tree with the localstorage button. From there, you can add a new sprite/object/transport to the level by clicking the + button on an existing entity, type in the needed bytes, and then add it. Unfortunately if a section is empty (like a room has no transports), then the + button won't show up. Hex-tree was totally just whipped together very quickly, it's pretty crappy.

Like this will add a new goomba to the level
![image](https://user-images.githubusercontent.com/141159/121247511-569db300-c870-11eb-94cd-f8dd6fcbf1d2.png)


I'll think about allowing people to add their own sprites to a level in smaghetti. The challenge with that is graphic sets, they really complicate sprites a lot. There's a high likelyhood if you manually added a sprite in smaghetti, it wouldn't show up correctly due to graphic sets. Maybe not that big of a deal since people using this feature are just experimenting.


also sprites need to be stored in the level file sorted by x. So if you add a sprite but its x position is not correct, the game will ignore it.
